### PR TITLE
Fix an intermittent launch message transfer error

### PR DESCRIPTION
**Pull request by tt-dma**
Issue here is that the launch.run field could be read as true before the launch.enable_brisc and launch.enable_ncrisc fields.
This fixes the intermittent issues with the DPrint RaiseWait test (passes 500 consecutive iterations, previously would fail within 100)

This PR also includes a couple of fixes I found when trying to get this new test to pass on CI.

### CODE REVIEW

File: tt_metal/impl/program/program.cpp [hunk 1]
```diff
@@ -610,14 +610,14 @@ void Program::compile( Device * device )
     }
 
     for (auto & f : events)
-        f.wait();
+        f.get();
```
Comment (tt-dma) [inline tt_metal/impl/program/program.cpp:613]: The problem I ran into here was that `future::wait()` discards any exceptions thrown by the event it's waiting for, meaning that a silent failure in a previous test that used the same kernel as my new test was causing only the new test to fail. I've confirmed `future::get()` does throw in the correct test.

**Comment by aliuTT** [comment]
Looks good to me!

File: tt_metal/impl/program/program.cpp (hunk 1 above)
Comment (aliuTT) [reply tt_metal/impl/program/program.cpp:613]: Is this the source of that CI map error?

Comment (tt-dma) [reply tt_metal/impl/program/program.cpp:613]: This + the fix in `runtime_args.cpp` were the cause of that CI map error

Comment (TT-billteng) [reply tt_metal/impl/program/program.cpp:613]: nice catch!

**Review by pgkeller — APPROVED** [review]

File: tests/tt_metal/tt_metal/unit_tests_frequent/module.mk [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+TT_METAL_UNIT_TESTS_FREQUENT_OBJS = $(addprefix $(OBJDIR)/, $(TT_METAL_UNIT_TESTS_FREQUENT_SRCS:.cpp=.o))
+TT_METAL_UNIT_TESTS_FREQUENT_DEPS = $(addprefix $(OBJDIR)/, $(TT_METAL_UNIT_TESTS_FREQUENT_SRCS:.cpp=.d))
+
+-include $(TT_METAL_UNIT_TESTS_FREQUENT_DEPS)
+
+# Each module has a top level target as the entrypoint which must match the subdir name
+tests/tt_metal/unit_tests_frequent: $(TESTDIR)/tt_metal/unit_tests_frequent
+
+.PRECIOUS: $(TESTDIR)/tt_metal/unit_tests_frequent
+$(TESTDIR)/tt_metal/unit_tests_frequent: $(TT_METAL_UNIT_TESTS_FREQUENT_OBJS) $(TT_METAL_LIB) $(TT_DNN_LIB)
```
Comment (tt-rkim) [inline tests/tt_metal/tt_metal/unit_tests_frequent/module.mk:24]: Get rid of `$(TT_DNN_LIB)` in this line

File: tests/tt_metal/tt_metal/unit_tests_frequent/module.mk [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Every variable in subdir must be prefixed with subdir (emulating a namespace)
+
+TT_METAL_UNIT_TESTS_FREQUENT_SRCS_HOME = tests/tt_metal/tt_metal/unit_tests_frequent
+
+TT_METAL_UNIT_TESTS_FREQUENT = ${TT_METAL_UNIT_TESTS_FREQUENT_SRCS_HOME}/tests_main.cpp
+TT_METAL_UNIT_TESTS_FREQUENT += $(wildcard ${TT_METAL_UNIT_TESTS_FREQUENT_SRCS_HOME}/*/*.cpp)
+TT_METAL_UNIT_TESTS_FREQUENT += $(wildcard ${TT_METAL_UNIT_TESTS_FREQUENT_SRCS_HOME}/*/*/*.cpp)
```
Comment (tt-rkim) [inline tests/tt_metal/tt_metal/unit_tests_frequent/module.mk:7]: Hmm... I'm wondering if you can just replace these 3 lines with:

`TT_METAL_UNIT_TESTS_FREQUENT += $(wildcard ${TT_METAL_UNIT_TESTS_FREQUENT_SRCS_HOME}/**/*.cpp)`

Could we give it a try?

**Review by tt-rkim — CHANGES_REQUESTED** [review]

File: tests/tt_metal/tt_metal/unit_tests_frequent/module.mk (hunk 2 above)
Comment (tt-dma) [reply tests/tt_metal/tt_metal/unit_tests_frequent/module.mk]: Removed in latest rebase

File: tests/tt_metal/tt_metal/unit_tests_frequent/module.mk (hunk 3 above)
Comment (tt-dma) [reply tests/tt_metal/tt_metal/unit_tests_frequent/module.mk]: Seems like it works, updated in latest rebase

**Review by tt-rkim — APPROVED** [review]
LGTM, as long as post commit passes and SD frequent shows your test being run

**Comment by tt-dma** [comment]
SD frequent shows the new test being run (and passing): https://github.com/tenstorrent-metal/tt-metal/actions/runs/7820474846/job/21335279279

**Review by TT-billteng — APPROVED** [review]

**Comment by tt-dma** [comment]
CI finally up and running, passing: https://github.com/tenstorrent-metal/tt-metal/actions/runs/7849471180

**Pull request merged**